### DISCUSSION TITLE
google crawling link in  __NEXT_DATA__

### DISCUSSION THREAD

**Discussion by nareshopoyi** [vercel/next.js#39377] [Help]
## Google crawling  and  SSR rendering

Might be i'm wrong but we are getting url hits which are actually exist in **\_\_NEXT_DATA__**  but not in HTML content.

ex. few data of  **\_\_NEXT_DATA__** 
1. "coverMedia":"/opoyi_ds5O6d3IC.jpg"  *(html render : /images/opoyi_ds5O6d3IC.jpg)*
2. "page":"/authors/[username]"

might be have more like this but we have noticed few of them in google search console
can you suggest  something regard this ?

**_Thanks_**

**Comment by zoul** [comment]
We have the same problem, and other people too – see [this Stack Overflow question](https://stackoverflow.com/questions/47210596/). It should be fixed on Google’s side, but I’m not holding my breath. Would it be possible to add some workaround to Next.js?

The Stack Overflow question suggests some tags to turn off Google indexing for given part of document, but [that doesn’t seem to be really supported by Google](https://twitter.com/JohnMu/status/1234952552141271046). Is there another way to prevent Google from indexing a part of the document?

Other than that I thought of some smudge/clean filter support that would scramble the `__NEXT_DATA__` contents in the source and descramble before using.

**Reply by nareshopoyi** [reply]
Thank you so much  Zoul, for yours reference this is really helpful

**Comment by filipeccm** [comment]
Hello, any update to this? We're having the same problem on Search Console

**Comment by geochanto** [comment]
Also running into this. Pretty big issue with no resolution that I can find anywhere.

**Comment by RhysSullivan** [comment]
Would adding rel="nofollow" to __NEXT_DATA__ solve this? There's not really any reason to have links in a script tag be followed especially when its used for hydration so I don't see any downside to this.

**Comment by hypervisor** [comment]
Having the same issue, would love a solution for this.

**Comment by RhysSullivan** [comment]
If any NextJS maintainers need access to search console + a NextJS project with this is occurring with I can provide access to both to debug it

**Comment by Blundert** [comment]
Same issue too :(

**Comment by phil-sola** [comment]
Same issue here. Is there any news or update on this? Seeing loads of image links in GSC and empty pages created on the site

**Comment by BleddP** [comment]
Still happening with the app directory too. Google is indexing images and links which are included in our API response from the CMS, but are not rendered in the HTML. They are still included in the `self.__next_f` scripts

**Comment by jan-bitgrip** [comment]
We have the same issue here at several enterprise client projects.  Thanks for the summary @zoul - our analysis and research had a very similar outcome.

Option 1:
Did someone checked or better verified the approach from @RhysSullivan with "adding rel="nofollow" to NEXT_DATA"? Would be an easy fix, if it works.

Edit: Not working, see https://github.com/vercel/next.js/issues/40143#issuecomment-1907789793

Option 2:
One workaround we read from and which should have an effect, would be to encode/decode all path like values in JSON data passed through components. But this would cause extra congnitive load for the application and is really ugly. But has someone a generic example for this one, maybe?

A NextJS native/generic solution would be great.

Here's a related issue: https://github.com/vercel/next.js/issues/40143
